### ISSUE TITLE
[a11y]: tag being interpreted as a group by voiceover

### ISSUE THREAD

**Issue by davidmenendez** [carbon-design-system/carbon#15885]
### Package

@carbon/react

### Package version

11.52

### Automated testing tool and ruleset

voiceover

### Assistive technology

voiceover

### Description

voiceover is describing tags as groups. this issue was originally brought up in the [carbon for ibm products repo](https://github.com/carbon-design-system/ibm-products/issues/4406).

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tag--default

### Steps to reproduce

view the [tag story in storybook](https://react.carbondesignsystem.com/?path=/story/components-tag--default), activate voiceover, and try to navigate from one tag to the next.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by davidmenendez** [comment]
i was able to replicate this locally and the issue appears to be related to usage of a `title` attribute in the underlying `span`. i don't believe a title is necessary here and removing it will alleviate the problem of voiceover recognizing a tag as a group.

**Comment by davidmenendez** [comment]
this seems like a good place to also ask this: what are carbon's requirements when testing voiceover? i asked initially in slack about this "problem" because in all honestly i don't know how much of a problem this is. this was brought to me by another developer with accessibility experience who identified the issue, but is it strictly a "problem" for voiceover to be identifying things like this as groups?

basically i just want to be sure that C4P is aligned with carbon in terms of accessibility testing so we can all be on the same page.

**Comment by davidmenendez** [comment]
@tw15egan thanks for reviewing. i did want to discuss the voiceover testing procedures for carbon, but if we want to take this offline or discuss in another issue please let me know 👍
